pkg/util/log/log.go

```diff
@@ -47,13 +47,30 @@ func InitLogger(logFormat string, logLevel dslog.Level, buffered bool, rateLimit
 		logger = log.With(logger, "ts", log.DefaultTimestampUTC, "caller", log.Caller(5))
 	}
 	// Must put the level filter last for efficiency.
-	logger = level.NewFilter(logger, logLevel.Option)
+	logger = newFilter(logger, logLevel)
 
 	// Set global logger.
 	Logger = logger
 	return logger
 }
 
+// Pass through Logger and implement the DebugEnabled interface that spanlogger looks for.
+type levelFilter struct {
```

Comment:
This looks worth to be moved to dskit too, so that other projects can use it too.